### PR TITLE
Firefly-92: more robust url creation

### PR DESCRIPTION
**Pull request by robyww**
- Better url creation
 - more robust url creation in `tellLabToLoadFileToServer`

### CODE REVIEW

File: src/FitsViewerExt.js [hunk 1]
```diff
@@ -178,13 +177,14 @@ export class FitsViewerWidget extends Widget {
...
-    const {fetchUrl}= firefly.util;
-    const UL_URL= window.document.location.href+ `/sendToFirefly?path=${filename}`;
-    const options = { method: 'GET' };
-    return fetchUrl(UL_URL,options,false, false)
-        .catch( e => {
-            console.log('Got Error from upload request');
+function tellLabToLoadFileToServer( path, firefly) {
+    const {searchParams,origin,pathname}= new URL(window.document.location.href);
+    searchParams.append('path',path);
+    const slashMaybe= pathname.endsWith('/') ? '' : '/';
```
Comment (stargaser) [inline src/FitsViewerExt.js:183]: For this part, I was expecting something that would truncate the starting url anywhere a question mark appears. I saw a suggestion to use `split('?')`.

File: src/FitsViewerExt.js [hunk 2]
```diff
@@ -178,13 +177,14 @@ export class FitsViewerWidget extends Widget {
...
+    const newURL= `${origin}${pathname}${slashMaybe}sendToFirefly?${searchParams.toString()}`;
```
Comment (stargaser) [inline src/FitsViewerExt.js:184]: Is there a standard way to join parts of a url? I didn't find one in a quick search. I saw a `url-join` package that is installable via npm.

**Review by stargaser — APPROVED** [review]
It works in standalone Jupyterlab. I expect it will work in Jupyterhub, but I wonder if there's a safer way to join the needed parts of the url (in line comments).

File: src/FitsViewerExt.js (hunk 1 above)
Comment (robyww) [reply src/FitsViewerExt.js:183]: URL handles that.

File: src/FitsViewerExt.js (hunk 2 above)
Comment (robyww) [reply src/FitsViewerExt.js:184]: I don't want to bring in extra npm packages. For just this. I am trying to stick more this standard JS. By using `origin` and searchParams the hard part is handled. The funny thing is that `URL` constructor should have been able to handle adding to the path if it ended with `/` but the `?` confuses it as well. I would actually have to do more parsing the make that work. I think this approach is the safest.

**Pull request merged**